### ISSUE TITLE
MAL XML Export Lists broken for both MAL and AniList

### ISSUE THREAD

**Issue by cluegithub** [Atelier-Shiori/Shukofukurou-iOS#32]
My primary device is running iOS 15 Developer Beta 4 (Build 19A5307g), and I have replicated this on an iPad running iOS 14.6. Attempting to export anime lists in MAL XML is broken. Exporting from AniList as source results in the error "One or more titles failed to export" and exporting from MAL as source results in an XML with null values for all entries and data points. This came after I reinstalled the app and rebooted my device multiple times attempting to fix a separate bug in which the app would crash when attempting to load my Kitsu account for an unknown reason; a sample crash log is attached. I accidentally tapped Kitsu in the Service selection menu (iOS 14.6) while attempting to diagnose this export issue and after fixing the crashing by reinstalling the app, the export functionality no longer works on iOS 14.6. After completely logging out of all accounts (iOS15), deleting Shukofukurou from the Storage settings menu, rebooting the device, and reinstalling Shukofukurou and logging into AniList, I was able to export some, but not all, of my stored titles to an XML.
Update: I stepped away from writing this for a while, and when I came back and attempted an export from AniList it appears to have successfully exported almost all my stored titles except a few. When I opened those titles in Title View before trying to export again it resolved the issue for those titles. I also resolved the Kitsu issue by deleting my Kitsu account, and now the app does not crash when navigating to Kitsu via the Services menu. Exporting from MAL still produces an XML with null values however.

[Shukofukurou-IOS-2021-08-03-172650.txt - Kitsu issue log](https://github.com/Atelier-Shiori/Shukofukurou-iOS/files/6927341/Shukofukurou-IOS-2021-08-03-172650.txt)
![AniList](https://user-images.githubusercontent.com/42914403/128102428-90bf1916-10e2-4338-a279-3bccf60ac4db.jpg)
![MAL](https://user-images.githubusercontent.com/42914403/128093268-77b61ffc-a609-4829-a995-f1c1dba27efc.jpg)

**Comment by moyitpro** [comment]
I find out the issues. It seems that when I implemented the XML exporter, the dictionary keys used to generate the XML are incorrect compared to the keys used for the entries used on the actual list. I will have a fix out sometime this weekend.

**Comment by moyitpro** [comment]
After some more debugging, the reason why the Kitsu and AniList exports fail is because the title id mapping service wasn"t operating properly, resulting in 500 Internal Server Error. I created a cron job to restart the service every week, so that shouldn"t happen.

I fixed the Myanimelist XML export with a recent commit and will release 2.1.10 with the fix sometime this weekend.

https://github.com/Atelier-Shiori/Shukofukurou-iOS/commit/0fa7f88ea24cfc910a405dfa5dfb99b703a960f0
